drivers/hmis/spec/requests/hmis/get_projects_spec.rb

```diff
@@ -0,0 +1,100 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'rails_helper'
+require_relative 'login_and_permissions'
+require_relative '../../support/hmis_base_setup'
+
+RSpec.describe Hmis::GraphqlController, type: :request do
+  before(:all) do
+    cleanup_test_environment
+  end
+  after(:all) do
+    cleanup_test_environment
+  end
+
+  include_context 'hmis base setup'
+
+  let!(:o2) { create :hmis_hud_organization, data_source: ds1, user: u1 }
+  let!(:p2) { create :hmis_hud_project, data_source: ds1, organization: o2, user: u1 } # This one shouldn't be viewable by this user
+  let!(:p3) { create :hmis_hud_project, project_name: 'TestProject - Name/with=special_chars', data_source: ds1, organization: o1, user: u1, project_type: 2 }
+
+  let(:query) do
+    <<~GRAPHQL
+      query GetProject($projectTypes: [ProjectType!], $searchTerm: String, $sortOrder: ProjectSortOption, $offset: Int = 0, $limit: Int = 5) {
+        projects(projectTypes: $projectTypes, searchTerm: $searchTerm, sortOrder: $sortOrder, offset: $offset, limit: $limit) {
+          nodesCount
+          nodes {
+            #{scalar_fields(Types::HmisSchema::Project)}
+          }
+        }
+      }
+    GRAPHQL
+  end
+
+  def search(**input)
+    aggregate_failures 'checking response' do
+      response, result = post_graphql(input) { query }
+      expect(response.status).to eq 200
+      project = result.dig('data', 'projects')
+      yield project
+    end
+  end
+
+  before(:each) do
+    hmis_login(user)
+    assign_viewable(view_access_group, o1, hmis_user)
+  end
+
+  it 'should return projects correctly without filters' do
+    search do |projects|
+      # Shouldn't return p2 since its org isn't viewable to us
+      expect(projects).to include(
+        'nodesCount' => 2,
+        'nodes' => contain_exactly(include('id' => p1.id.to_s), include('id' => p3.id.to_s)),
+      )
+    end
+  end
+
+  it 'should return projects correctly with project type filter' do
+    search(project_types: [Types::HmisSchema::Enums::ProjectType.key_for(p3.project_type)]) do |projects|
+      expect(projects).to include(
+        'nodesCount' => 1,
+        'nodes' => contain_exactly(include('id' => p3.id.to_s)),
+      )
+    end
+  end
+
+  describe 'with search term' do
+    [
+      [
+        'project', # Search term
+        ->(_ctx) { [p1, p3] }, # Expected results
+        ->(_ctx) { p1.update(project_name: 'Project 1') }, # Test setup (optional)
+      ],
+      ['TestProject - ', ->(_ctx) { [p3] }],
+      ['TestProject - Name/with=special_chars', ->(_ctx) { [p3] }],
+      ['test name special', ->(_ctx) { [p3] }],
+      # Add more search term tests here
+    ].each do |term, get_expected_projects, setup|
```

Comment:
I'm not a _huge_ fan of the complexity here, but I can follow it, so it's probably fine.  It _might_ be easier to follow if this was array of hashes:
```suggestion
    [
      {
        term: 'project', # Search term
        result_lamda: -> { [p1, p3] }, # Expected results
        setup_lamda: -> { p1.update(project_name: 'Project 1') }, # Test setup (optional)
      },
      { term: 'TestProject - ', result_lamda: -> { [p3] }},
      { term: 'TestProject - Name/with=special_chars', result_lamda: -> { [p3] }},
      { term: 'test name special', result_lamda: -> { [p3] }},
      # Add more search term tests here
    ].each do |test_data|
```

Then you can reference the object with something like:
```
test_data[:setup_lamda].call
```

Reply:
Gotcha. On second thought I think I can probably just put all of those into a single `it` rather than create an `it` for each search term. That should eliminate the complexity of having to do the lambdas and probably make it more readable. What do you think of that?

Reply:
Sure, individual tests is totally fine!